libs/animation/animation.ts

```diff
@@ -340,11 +357,15 @@ namespace animation {
             super();
         }
 
-        apply(target: Sprite, nodeTime: number, interval: number) {
+        apply(target: Sprite, nodeTime: number, interval: number): boolean {
             const x = Math.round(((this.p1.x - this.p0.x) / interval) * nodeTime) + this.p0.x;
             const y = Math.round(((this.p1.y - this.p0.y) / interval) * nodeTime) + this.p0.y;
-
+            if (!isClearPath(interval, target, target.x, target.y, x, y))
+            {
```

Comment:
style: we put open curly braces on the same lines as the if

Reply:
Fixed!